### PR TITLE
Add rdi to address record

### PR DESCRIPTION
**Pull request by nononoy**
When verifying addresses and if  they are valid we get an error: 

```
TestAddress#test_verify_good_address_with_rdi_returned:
NoMethodError: undefined method `rdi=' for #<TaxCloud::Address:0x00007f27a7922ea0 @address1="888 6th Ave", @city="New York", @state="NY", @zip5="10001", @zip4="3502">

<s:complexType name="VerifiedAddress">
  <s:complexContent mixed="false">
    <s:extension base="tns:Address">
      <s:sequence>
        <s:element minOccurs="0" maxOccurs="1" name="rdi" type="s:string"/>
        <s:element minOccurs="0" maxOccurs="1" name="ErrNumber" type="s:string"/>
        <s:element minOccurs="0" maxOccurs="1" name="ErrDescription" type="s:string"/>
      </s:sequence>
    </s:extension>
  </s:complexContent>
</s:complexType>
```

The issue is blocking so it'd be great to fix it asap

### CODE REVIEW

**Review by dblock — CHANGES_REQUESTED** [review]
Looks good!

1. Bump version to 0.6.0 in this PR.
2. Add a line to https://github.com/txcrb/tax_cloud/blob/master/CHANGELOG.md
3. Instead of replacing the recorded tax cloud responses (the recorded .yml), add new ones with RDI and add separate tests for RDI only. This way we know we didn't break anything.

File: lib/tax_cloud/address.rb [hunk 1]
```diff
@@ -15,6 +15,8 @@ class Address < Record
     attr_accessor :zip5
     # 4-digit Zip Code.
     attr_accessor :zip4
+    # RDI
```
Comment (dblock) [inline lib/tax_cloud/address.rb:18]: Nit: let's expand what RDI means here?

Comment (nononoy) [reply lib/tax_cloud/address.rb]: I suppose it's `Residential Delivery Indicator`

**Comment by nononoy** [comment]
I'm not sure, that keeping old invalid responses could save us from mistakes. It is better to refresh them all and update the code accordingly, since versioning is unavailable on the tax_cloud side. Also, I think, that instead of removing attributes we should whitelist them to avoid such issues in the future.
```
result.delete(:err_number) -> result.slice(TaxCloud::Address.attributes) or something
```
I have some spare time to invest, so ping me if we're on the same page.

**Comment by dblock** [comment]
@nononoy I'm with you!

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
-### 0.5.1 (Next)
+### 0.6.0 (12/13/2023)
```
Comment (dblock) [inline CHANGELOG.md:1]: For now write "Next", we'll release and change it.

**Review by dblock — APPROVED** [review]
Looks good. Remove the date to avoid confusion?

File: CHANGELOG.md (hunk 2 above)
Comment (nononoy) [reply CHANGELOG.md]: sure, done

**Comment by dblock** [comment]
I released 0.6.0.

**Comment by nononoy** [comment]
@dblock thank you!

**Pull request merged**